Source/WebCore/html/PermissionsPolicy.cpp

```diff
@@ -413,29 +451,27 @@ bool PermissionsPolicy::inheritedPolicyValueForFeature(Feature feature) const
 // https://w3c.github.io/webappsec-permissions-policy/#algo-create-for-navigable
 PermissionsPolicy::PermissionsPolicy(const HTMLFrameOwnerElement* frame, const SecurityOriginData& origin)
 {
-    m_inheritedPolicy.set(index(Feature::Camera), computeInheritedPolicyValueInContainer(Feature::Camera, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Microphone), computeInheritedPolicyValueInContainer(Feature::Microphone, frame, origin));
-    m_inheritedPolicy.set(index(Feature::SpeakerSelection), computeInheritedPolicyValueInContainer(Feature::SpeakerSelection, frame, origin));
-    m_inheritedPolicy.set(index(Feature::DisplayCapture), computeInheritedPolicyValueInContainer(Feature::DisplayCapture, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Gamepad), computeInheritedPolicyValueInContainer(Feature::Gamepad, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Geolocation), computeInheritedPolicyValueInContainer(Feature::Geolocation, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Payment), computeInheritedPolicyValueInContainer(Feature::Payment, frame, origin));
-    m_inheritedPolicy.set(index(Feature::ScreenWakeLock), computeInheritedPolicyValueInContainer(Feature::ScreenWakeLock, frame, origin));
-    m_inheritedPolicy.set(index(Feature::SyncXHR), computeInheritedPolicyValueInContainer(Feature::SyncXHR, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Fullscreen), computeInheritedPolicyValueInContainer(Feature::Fullscreen, frame, origin));
-    m_inheritedPolicy.set(index(Feature::WebShare), computeInheritedPolicyValueInContainer(Feature::WebShare, frame, origin));
-#if ENABLE(DEVICE_ORIENTATION)
-    m_inheritedPolicy.set(index(Feature::Gyroscope), computeInheritedPolicyValueInContainer(Feature::Gyroscope, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Accelerometer), computeInheritedPolicyValueInContainer(Feature::Accelerometer, frame, origin));
-    m_inheritedPolicy.set(index(Feature::Magnetometer), computeInheritedPolicyValueInContainer(Feature::Magnetometer, frame, origin));
-#endif
-#if ENABLE(WEB_AUTHN)
-    m_inheritedPolicy.set(index(Feature::PublickeyCredentialsGetRule), computeInheritedPolicyValueInContainer(Feature::PublickeyCredentialsGetRule, frame, origin));
-#endif
-#if ENABLE(WEBXR)
-    m_inheritedPolicy.set(index(Feature::XRSpatialTracking), computeInheritedPolicyValueInContainer(Feature::XRSpatialTracking, frame, origin));
-#endif
-    m_inheritedPolicy.set(index(Feature::PrivateToken), computeInheritedPolicyValueInContainer(Feature::PrivateToken, frame, origin));
+    for (size_t index = 0; index < numberOfFeatures; ++index)
+        m_inheritedPolicy.set(index, computeInheritedPolicyValueInContainer(feature(index), frame, origin));
```

Comment:
I am not sure I like BitSet, it removes enum typing for instance and makes the code less readable.
We could have a simple struct of booleans that would be as compact as a bitset.
And utility routines like `forEach(const Function<void(Feature)>&)` might make it convenient enough so that adding a new Feature will be only a few lines of code.